src/db/tables/email_addresses.js

```diff
@@ -54,6 +54,20 @@ async function addSubscriberUnverifiedEmailHash (user, email) {
 
 async function resetUnverifiedEmailAddress (emailAddressId) {
   const newVerificationToken = uuidv4()
+
+  // Time in ms to require between verification reset.
+  const verificationWait = 5 * 60 * 1000 // 5 minutes
+
+  const verificationRecentlyUpdated = await knex('email_addresses')
+    .select('id')
+    .whereRaw(`"updated_at" > NOW() - INTERVAL '${verificationWait} MILLISECONDS'`)
+    .andWhere('id', emailAddressId)
+    .first()
+
+  if (verificationRecentlyUpdated?.id === parseInt(emailAddressId)) {
```

Comment:
It would be nice if knex supported `SELECT EXISTS` for this kind of thing, but we'd have to use a completely raw query and I'm trying to minimize that.